src/index.ts

```diff
@@ -0,0 +1,169 @@
+            return res
+              .writeHead(201, "Created")
+              .end(`Data has been successfully written to file`);
+          });
+        });
+      });
+      break;
+    case "/put":
+      res.setHeader("content-type", "text/plain");
+
+      if (req.method !== "PUT") {
+        return res.writeHead(403, "Forbidden").end(`Error 403. Forbidden`);
+      }
+
+      let bodyOfPut = "";
+      req.on("data", (chunk) => {
+        bodyOfPut += chunk.toString();
+      });
+
+      req.on("end", () => {
+        const parsedBody: ParsedQuery<string> = parse(bodyOfPut);
+
+        if (!parsedBody.id || !parsedBody.name || !parsedBody.age) {
+          return res
+            .writeHead(400, "Bad Request")
+            .end(`Error 400. Bad Request`);
+        }
+
+        fs.readFile(pathToJsonFile, "utf-8", (err, content) => {
+          if (err) {
+            return res
+              .writeHead(500, "Internal Server Error")
+              .end(`Error 500. Internal Server Error`);
+          }
+
+          const users: UsersType = JSON.parse(content);
+
+          const changedUser: typeOfUser = {
+            id: Number(parsedBody.id) || 0,
+            name: parsedBody.name?.toString() || "",
+            age: parsedBody.age?.toString() || "",
+          };
+          const newListOfUsers = users.map((user) => {
+            if (user.id === changedUser.id) {
+              return changedUser;
+            }
+            return user;
+          });
+
+          fs.writeFile(pathToJsonFile, JSON.stringify(newListOfUsers), () => {
+            return res
+              .writeHead(201, "Created")
+              .end(`Data has been successfully changed`);
+          });
+        });
+      });
+      break;
+    case "/delete":
+      res.setHeader("content-type", "text/plain");
+      if (req.method !== "DELETE") {
+        return res.writeHead(403, "Forbidden").end(`Error 403. Forbidden`);
+      }
+
+      let bodyOfDelete = "";
+      req.on("data", (chunk) => {
+        bodyOfDelete += chunk.toString();
+      });
+
+      req.on("end", () => {
+        const parsedBody: ParsedQuery<string> = parse(bodyOfDelete);
+
+        if (!parsedBody.id) {
+          return res
+            .writeHead(400, "Bad Request")
+            .end(`Error 400. Bad Request`);
+        }
+
+        fs.readFile(pathToJsonFile, "utf-8", (err, content) => {
+          if (err) {
+            return res
+              .writeHead(500, "Internal Server Error")
+              .end(`Error 500. Internal Server Error`);
+          }
+
+          const users: UsersType = JSON.parse(content);
+
+          const filteredUsers = getFilteredUsers(
+            users,
+            parsedBody.id?.toString() || ""
+          );
+
+          fs.writeFile(pathToJsonFile, JSON.stringify(filteredUsers), () => {
+            return res
+              .writeHead(200, "Ok")
+              .end(`Data has been successfully deleted`);
+          });
```

Comment:
Hmm, it's not a DELETE operation. It's just update. For delete operation you should remove file. We work with entities. File content is not an entity.